src/julia.jl

```diff
@@ -33,4 +33,139 @@ function multireshape(parent::AbstractArray, dimss::Array{Tuple{Vararg{Int}}})
     multireshape(parent, dimss...)
 end
 
+
+"""
+    pairwise(f::Function, a; kwargs...)
+
+Apply the function `f` to all pairwise combinations of elements from `a`. Set
+keyword `symmetric = true` to only calculate `f(x,y)` when `f(x,y) == f(y,x)`.
+"""
+function pairwise(f::Function, a; symmetric = false)
+    n = length(a)
+    f = tryfunc(f, a[1], a[1])
+    first = f(a[1], a[1])
+    r = Matrix{typeof(first)}(n, n)
+    pairwise!(f, a, r; symmetric = symmetric)
+    r
+end
+
+"""
+    pairwise(f::Function, a, output; kwargs...)
+
+Apply the function `f` to all pairwise combinations of elements from `a`, and
+return a `Vector` of the results if `output = :Vector`. The Vector form is
```

Comment:
Right, in this case I see that passing the type `Vector` is a better way of doing it thanks. And strictly speaking it is not necessary that we discuss it, as this function would be removed with my suggested edit to the PR below (I will wait for you or @ChrisRackauckas to comment on that before implementing it).
But – as a learning process I think it would be nice to understand why this is "very weird and un-Julian"? I got the idea from this bit of code that I contributed to GLM.jl: https://github.com/JuliaStats/GLM.jl/blob/master/src/lm.jl#L159
When I made the PR I was worried about type stability when returning confidence intervals because that would require returning a matrix rather than a vector, and nalimilan suggested this solution with having a Symbol argument specifying the type of confidence interval (so far only :confint is implemented, but :predint is planned) – I thought it was elegant and so added it to my repertoire.